marquez/client.py

```diff
@@ -0,0 +1,264 @@
+    def get_datasource(self, urn, **kwargs):
+        path = "datasources/{0}"
+        path_args = [urn]
+        if 'timeout' not in kwargs:
+            kwargs['timeout'] = self.timeout
+        return request_utils.get_request(
+            self.marquez_host, path, path_args, **kwargs)
+
+    @handle_response
+    def create_datasource(self, name, connection_url, **kwargs):
+        path = "datasources"
+        path_args = []
+        payload = {
+            "name": name,
+            "connectionUrl": connection_url
+        }
+        if 'timeout' not in kwargs:
+            kwargs['timeout'] = self.timeout
+        return request_utils.post_request(
+            self.marquez_host, path, path_args, payload, **kwargs)
+
+    #  Jobs API
+    @handle_response
+    def create_job(self, job_name, location, input_dataset_urns,
+                   output_dataset_urns, description=None, **kwargs):
+        path = "namespaces/{0}/jobs/{1}"
+        path_args = [self.namespace, job_name]
+        payload = {
+            "inputDatasetUrns": input_dataset_urns,
+            "outputDatasetUrns": output_dataset_urns,
+            "location": location
+        }
+
+        if description:
+            payload['description'] = description
+        if 'timeout' not in kwargs:
+            kwargs['timeout'] = self.timeout
+        return request_utils.put_request(
+            self.marquez_host, path, path_args, payload, **kwargs)
+
+    @handle_response
+    def get_job(self, job_name, **kwargs):
+        path = "namespaces/{0}/jobs/{1}"
+        path_args = [self.namespace, job_name]
+        if 'timeout' not in kwargs:
+            kwargs['timeout'] = self.timeout
+        return request_utils.get_request(
+            self.marquez_host, path, path_args, **kwargs)
+
+    @handle_response
+    def get_jobs(self, **kwargs):
+        path = "namespaces/{0}/jobs/"
+        path_args = [self.namespace]
+        if 'timeout' not in kwargs:
+            kwargs['timeout'] = self.timeout
+        return request_utils.get_request(
+            self.marquez_host, path, path_args, **kwargs)
+
+    @handle_response
+    def create_job_run(self, job_name, job_run_args,
+                       nominal_start_time=None,
+                       nominal_end_time=None, **kwargs):
+        path = "namespaces/{0}/jobs/{1}/runs"
+        path_args = [self.namespace, job_name]
+        payload = {
+            "nominalStartTime": nominal_start_time,
+            "nominalEndTime": nominal_end_time,
+            "runArgs": job_run_args
+        }
+        if 'timeout' not in kwargs:
+            kwargs['timeout'] = self.timeout
+        return request_utils.post_request(
+            self.marquez_host, path, path_args, payload, **kwargs)
+
+    @handle_response
+    def get_job_run(self, jobrun_id, **kwargs):
+        path = "jobs/runs/{0}"
+        path_args = [jobrun_id]
+
+        if 'timeout' not in kwargs:
+            kwargs['timeout'] = self.timeout
```

Comment:
this is repeated lots of times

```
        if 'timeout' not in kwargs:
            kwargs['timeout'] = self.timeout
```

can this just be pushed down into the low-level request methods?

Reply:
Did some refactoring and was able to push it down.